compiler/semexprs.nim

```diff
@@ -2357,14 +2362,6 @@ proc semTuplePositionsConstr(c: PContext, n: PNode, flags: TExprFlags): PNode =
     addSonSkipIntLit(typ, n.sons[i].typ)
   result.typ = typ
 
-proc isTupleType(n: PNode): bool =
```

Comment:
I dislike your inlining of this nice helper. Instead make it return an enum.

Reply:
Well subjectively speaking, I do like to have this helper inlined a lot more. It allows me to call localError and set the error node. I don't need to introduce yet another enum that is just just as a result type for this one function that isn't even used in more than one place. I even think it is more readable this way. But you really really want this "nice" helper?

Reply:
Alternatively move this ``of`` branch into a ``semTupleConstr`` proc.